orchagent/neighorch.cpp

```diff
@@ -428,6 +428,35 @@ void NeighOrch::doTask(Consumer &consumer)
 
         IpAddress ip_address(key.substr(found+1));
 
+        /* The following change is based on the following PR:
+         * https://github.com/Azure/sonic-swss/pull/1431
+         * 
+         * For inband Vlan, a kernel neigh is added for every remote neighbor
+         * and the interface that the neigh is attached is inband Vlan.
+         * neighsyncd adds the kernel neigh to neigh table in APPL_DB again.
+         * The following change is to skip adding these remote neighbor attached
+         * to inband Vlan.
+         *
+         * The collection m_remoteNeigh is added to track remote neighbor. It's
+         * updated in the following method introduced in the above PR:
+         *  
+         * doVoqSystemNeighTask()
+         */
+        if (gPortsOrch->isInbandPort(alias))
+        {
+            Port ibport;
+            gPortsOrch->getInbandPort(ibport);
+            if(ibport.m_type == Port::VLAN)
+            {
+                 /* Only create neighbor attached to inband Vlan */
+                if (m_remoteNeigh.find(ip_address) != m_remoteNeigh.end())
```

Comment:
Yeah.  iBGP peer can be resolved over inband Vlan. This kind of inband vlan neighbor needs to be created. But remote neighbors can be skipped.

Reply:
@vganesan-nokia why same cannot happen with Inband port ? iBGP neighbor can happen on inband port also ?

Reply:
The only neighbors that would be available on the inband port are the static neigbors we add for the remote neighbors (which should be skipped here since these are already added to SAI as part of doVoqSystemNeighTask()). Unlike inband vlan, inband port will not have any other neighbors. So no need to collect or verify whether the neihbor is remote or not. Just check the interface of the neighbor and if the interface is inband interface, skip the neighbor. 

For inband port except the ip address of inband port of my asic, all the iBGP neighbors (which are ip address of inband port of other asics) are remote neighbors. These are already programmed in SAI by doVoqSystemNeighTask() so we skip here. But inband vlan will have both the iBGP neighbors (resolved via ARP) which should be programmed  here and remote neighbors which are added statically to inband vlan and SAI prgrammed in doVoqSystemNeighTask(). Therefore for inband vlan case, we need to indentify the remote neighbors and skip.